crates/evm/evm.rs

```diff
@@ -300,7 +300,7 @@ pub fn beacon_root_contract_call(
     let tx_env = TxEnv {
         caller: *SYSTEM_ADDRESS,
         transact_to: RevmTxKind::Call(*CONTRACT_ADDRESS),
-        gas_limit: 30_000_000,
+        gas_limit: header.gas_limit,
```

Comment:
This should be set to 30_000_000 according to the EIP 4788 spec:
https://eips.ethereum.org/EIPS/eip-4788#block-processing

Reply:
We get the error "caller gas limit exceeds the block gas limit" from the evm with that value. (This comes from the check `self.tx.gas_limit > self.block.gas_limit` in `validate_tx`)

Reply:
We could also fix it by setting the block_env's gas_limit to 30_000_000 too

Reply:
Update: This last solution works fine when running with kurtosis, I pushed it